tests/test_standardizer.py

```diff
@@ -0,0 +1,357 @@
+            KBMODV1(**standardized["meta"], force=KBMODV1)
+
+        # Test it works correctly when the FITS is reachable.
+        fits_file = tempfile.NamedTemporaryFile(suffix=".fits", delete=False)
+        self.fits.writeto(fits_file.file, overwrite=True, output_verify="ignore")
+        fits_file.close()
+
+        std = Standardizer.get(fits_file.name, force=KBMODV1)
+        standardized = std.standardize()
+        std2 = KBMODV1(**standardized["meta"])
+        self.assertIsInstance(std2, KBMODV1)
+
+    def test_bitmasking(self):
+        """Test masking with direct config works as expected."""
+        # Assign each flag that exists to a pixel, standardize, then expect
+        # the mask to only contain those pixels that are also in mask_flags.
+        # The grow_kernel is so large by default it would mask the nearly the
+        # whole image, so we turn it off.
+        KBMODV1Config.grow_mask = False
+        mask_arr = self.fits["MASK"].data
+        for i, flag in enumerate(KBMODV1Config.bit_flag_map):
+            mask_arr.ravel()[i] = KBMODV1Config.bit_flag_map[flag]
+
+        std = Standardizer.get(self.fits, force=KBMODV1)
+        standardizedMask = std.standardizeMaskImage()
+
+        for mask in standardizedMask:
+            for i, flag in enumerate(KBMODV1Config.bit_flag_map):
+                with self.subTest("Failed to mask expected", flag=flag):
+                    if flag in KBMODV1Config.mask_flags:
+                        self.assertEqual(mask.ravel()[i], True)
+                    else:
+                        self.assertEqual(mask.ravel()[i], False)
+
+    def test_threshold_masking(self):
+        """Test brightness threshold masking. Test config overrides."""
+        # set one pixel that is masked and one that isn't
+        self.fits["IMAGE"].data[1, 1] = 1
+        self.fits["IMAGE"].data[2, 2] = 3
+
+        conf = StandardizerConfig(
+            {
+                "grow_mask": False,
+                "do_threshold": True,
+                "brightness_threshold": 2,
+            }
+        )
+        std = Standardizer.get(self.fits, force=KBMODV1, config=conf)
+        mask = next(std.standardizeMaskImage())
+
+        self.assertFalse(mask[1, 1])
+        self.assertTrue(mask[2, 2])
+
+    def test_grow_mask(self):
+        """Test mask grows as expected."""
+        # set central pixel to be masked, then grow that mask to all its
+        # neighbors.
+        self.fits["MASK"].data[2, 2] = KBMODV1Config.bit_flag_map["BAD"]
+
+        conf = StandardizerConfig({"grow_mask": True, "grow_kernel_shape": (3, 3)})
+        std = Standardizer.get(self.fits, force=KBMODV1, config=conf)
+        mask = next(std.standardizeMaskImage())
+
+        # Note that this is different than masking via Manhattan neighbors -
+        # which can be implemented by using the C++ functions in KBMODV1, do I?
+        # the solution now is an masked square in the center of the array
```

Comment:
I think we asserted that it likely won't matter practically. So we can leave this approach for now and check if we see a massive drop in the number of candidate tracks.

Reply:
Right, I feel like a lot of work went into them so I'm loath about just deleting them. Much of the C_++ functionality would work just fine as a C++ wrapped python function, would likely be faster than what we have, and make our lives easier long-term too.

It would be kind of hard to stuff the current masking setup into standardizers so I appreciate the flexibility and patience!